File: Easy/Skynet the Chasm/C++/Skynet the Chasm.cpp
#include <iostream>

using namespace std;

int main()
{
	int lenghtOfRoad;
	cin >> lenghtOfRoad; cin.ignore();
	int lenghtOfGap;
	cin >> lenghtOfGap; cin.ignore();
	int lenghtOfPlatform;
	cin >> lenghtOfPlatform; cin.ignore();

	while (1) {
		int motorbikeSpeed;
		cin >> motorbikeSpeed; cin.ignore();
		int motorbikePosition;
		cin >> motorbikePosition; cin.ignore();

		cout <<
			(motorbikePosition > lenghtOfRoad || motorbikeSpeed > lenghtOfGap + 1 ? "SLOW" :
			motorbikeSpeed < lenghtOfGap + 1 ? "SPEED" :
			motorbikePosition == lenghtOfRoad - 1 ? "JUMP" : "WAIT")
			<< endl;
	}
}

File: Easy/Skynet the Chasm/PHP/SkynetTheChasm.php
<?php
fscanf(STDIN, "%d", $lenghtOfRoad);
fscanf(STDIN, "%d", $lenghtOfGap);
fscanf(STDIN, "%d", $lenghtOfPlatform);

while (TRUE)
{
    fscanf(STDIN, "%d", $motorbikeSpeed);
    fscanf(STDIN, "%d", $motorbikePosition);

    if ($motorbikePosition > $lenghtOfRoad || $motorbikeSpeed > $lenghtOfGap + 1) {
        echo "SLOW\n";
    } elseif ($motorbikeSpeed < $lenghtOfGap + 1) {
        echo "SPEED\n";
    } elseif ($motorbikePosition == $lenghtOfRoad - 1) {
        echo "JUMP\n";
    } else {
        echo "WAIT\n";
    }
}
?>